can-construct_test.js

```diff
@@ -159,3 +159,44 @@ test("setup called with original arguments", function(){
 
 	Construct.extend(o1, o2);
 });
+
+test("legacy namespace strings (A.B.C) accepted", function() {
+
+	var Type = Construct.extend("Foo.Bar.Baz");
+
+	ok(new Type() instanceof Construct, "No unexpected behavior in the prototype chain");
+	equal(Type.name, "Foo_Bar_Baz", "Name becomes underscored");
```

Comment:
Changes made.  Assuming travis passes this should be ready for final review and passing.

It's not outlined in the test but I did do a manual test battery to ensure that caching the function constructors didn't cause collisions when two constructors had the same name.